### PR TITLE
Fixing issue with casing on operations `wit_update_work_item` rejects…

### LINKED ISSUE #323: `wit_update_work_item` rejects lowercase operation values ("add", "replace", "remove"): returns error `invalid_enum_value`
When I pass an update work item states to MCP’s wit_update_work_item tool using a lowercase operation value ("add"), the call fails with error -32602 invalid_enum_value. The tool only accepts the capital‑cased enum set ("Add" | "Replace" | "Remove"). This behavior is unexpected because the documentation and most JSON‑Patch conventions treat the op field as case‑insensitive.

mcp input
```
{
  "id": 39,
  "updates": [
    {
      "op": "add",
      "path": "/fields/System.State",
      "value": "Resolved"
    }
  ]
}
```

mcp output
```
Error: MCP -32602: MCP error -32602: Invalid arguments for tool wit_update_work_item: [
  {
    "received": "add",
    "code": "invalid_enum_value",
    "options": [
      "Add",
      "Replace",
      "Remove"
    ],
    "path": [
      "updates",
      0,
      "op"
    ],
    "message": "Invalid enum value. Expected 'Add' | 'Replace' | 'Remove', received 'add'"
  }
]
```

Thanks for your attention! Let me know if I can provide additional logs or help test a fix.

### PR DESCRIPTION
**Pull request by danhellem**
This pull request refactors the handling of work item update operations in `src/tools/workitems.ts` and enhances the corresponding test coverage in `test/src/tools/workitems.test.ts`. Key changes include simplifying the transformation of operation strings, removing redundant code, and adding tests for edge cases like API failures, operation transformations, and Markdown formatting.

### Refactoring and simplification:

* Removed the `operationToApiString` utility function and replaced it with an inline transformation using `zod` schemas to convert operation strings to lowercase and validate them against allowed values (`add`, `replace`, `remove`) (`src/tools/workitems.ts`, [[1]](diffhunk://#diff-86312c74c8d340f1b252bb6a34ae3d610c400cf9151d45223ec54d2d2ab2b0c9L9-L21) [[2]](diffhunk://#diff-86312c74c8d340f1b252bb6a34ae3d610c400cf9151d45223ec54d2d2ab2b0c9L466-R461).
* Simplified the mapping of update operations by directly using the transformed `op` values instead of calling the removed utility function (`src/tools/workitems.ts`, [src/tools/workitems.tsL480-R475](diffhunk://#diff-86312c74c8d340f1b252bb6a34ae3d610c400cf9151d45223ec54d2d2ab2b0c9L480-R475)).

### Enhanced test coverage:

* Added a test case to ensure proper error handling when the API fetch call fails, verifying that an appropriate error message is thrown (`test/src/tools/workitems.test.ts`, [test/src/tools/workitems.test.tsR430-R456](diffhunk://#diff-81b8c1c7196cc4eba4c6b2a30eaec7c3101f72c5a88803d07b942b71e3404b9dR430-R456)).
* Extended tests to validate the transformation of uppercase operation strings (e.g., `Add`, `Replace`) to lowercase and their correct usage in API calls (`test/src/tools/workitems.test.ts`, [test/src/tools/workitems.test.tsL608-R653](diffhunk://#diff-81b8c1c7196cc4eba4c6b2a30eaec7c3101f72c5a88803d07b942b71e3404b9dL608-R653)).
* Introduced a test to verify that updates are grouped by work item ID when batching API requests (`test/src/tools/workitems.test.ts`, [test/src/tools/workitems.test.tsR975-R1060](diffhunk://#diff-81b8c1c7196cc4eba4c6b2a30eaec7c3101f72c5a88803d07b942b71e3404b9dR975-R1060)).
* Added a test to ensure that long text fields are correctly formatted as Markdown when specified, including verifying the addition of a `multilineFieldsFormat` entry in the API request body (`test/src/tools/workitems.test.ts`, [test/src/tools/workitems.test.tsR975-R1060](diffhunk://#diff-81b8c1c7196cc4eba4c6b2a30eaec7c3101f72c5a88803d07b942b71e3404b9dR975-R1060)).

Fixes #323

## **Associated Risks**

No risks

## ✅ **PR Checklist**

- [x] **I have read the [contribution guidelines](https://github.com/microsoft/azure-devops-mcp/blob/main/CONTRIBUTING.md)**
- [x] **I have read the [code of conduct guidelines](https://github.com/microsoft/azure-devops-mcp/blob/main/CODE_OF_CONDUCT.md)**
- [x] Title of the pull request is clear and informative.
- [x] 👌 Code hygiene
- [x] 🔭 Telemetry added, updated, or N/A
- [x] 📄 Documentation added, updated, or N/A
- [x] 🛡️ Automated tests added, or N/A

## 🧪 **How did you test it?**

Updated auto tests. Verified the tools works manually and updated cases on operations to account for uppper, lower, and combo

### CODE REVIEW

File: test/src/tools/workitems.test.ts [hunk 1]
```diff
@@ -605,16 +632,25 @@ describe("configureWorkItemTools", () => {
         id: 131489,
         updates: [
           {
-            op: "add",
+            op: "Add", // Uppercase to test transformation to lowercase (line 456)
```
Comment (egor-bryzgalov) [inline test/src/tools/workitems.test.ts]: nitpick: referring to a line can be outdated quite fast, should we remove the comment or at least remove the reference?

File: src/tools/workitems.ts [hunk 2]
```diff
@@ -463,7 +453,12 @@ function configureWorkItemTools(server: McpServer, tokenProvider: () => Promise<
       updates: z
         .array(
           z.object({
-            op: z.enum(["Add", "Replace", "Remove"]).default("Add").describe("The operation to perform on the field."),
+            op: z
+              .string()
+              .transform((val) => val.toLowerCase())
```
Comment (aaudzei) [inline src/tools/workitems.ts:458]: is the transformation necessary considering the `enum(["add", "replace", "remove"]` below is already a selection of lowercased options?

Comment (danhellem) [reply src/tools/workitems.ts:458]: The issue here is that people can send "add", or "ADD" or "Add".  Previously the enum was set to "Add" and that was the only thing accepted. But in the REST API call we us all lower case. This fix allows them send whatever casing they want and we just auto convert to lower.

File: test/src/tools/workitems.test.ts [hunk 3]
```diff
@@ -936,6 +972,92 @@ describe("configureWorkItemTools", () => {
 
       const result = await handler(params);
 
+      // This verifies that the updates are grouped by work item ID as implemented in line 643
```
Comment (egor-bryzgalov) [inline test/src/tools/workitems.test.ts:975]: nitpick: i think we should remove line references from all comments
